notify/impl.go

```diff
@@ -264,7 +264,16 @@ func (n *Email) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
 	}
 
 	if port == "465" {
-		conn, err := tls.Dial("tcp", n.conf.Smarthost, &tls.Config{ServerName: host})
+
+		if n.conf.TLSConfig.ServerName == "" {
+			n.conf.TLSConfig.ServerName = host
```

Comment:
This modifies the original configuration object and would be reflected in `/api/v1/status` (or the Status page). Better to do it after `commoncfg.NewTLSConfig(...)`.